server/services/ManagedIndexService.ts

```diff
@@ -31,34 +31,50 @@ import {
 } from "../models/interfaces";
 import { ManagedIndicesSort, ServerResponse } from "../models/types";
 
-type Request = Legacy.Request;
-type ElasticsearchPlugin = Legacy.Plugins.elasticsearch.Plugin;
-type ResponseToolkit = Legacy.ResponseToolkit;
-
 export default class ManagedIndexService {
-  esDriver: ElasticsearchPlugin;
+  esDriver: IClusterClient;
 
-  constructor(esDriver: ElasticsearchPlugin) {
+  constructor(esDriver: IClusterClient) {
     this.esDriver = esDriver;
   }
 
   // TODO: Not finished, need UI page that uses this first
-  getManagedIndex = async (req: Request, h: ResponseToolkit): Promise<ServerResponse<any>> => {
+  getManagedIndex = async (
+    context: RequestHandlerContext,
+    request: KibanaRequest,
+    response: KibanaResponseFactory
+  ): Promise<IKibanaResponse<ServerResponse<any>>> => {
     try {
-      const { id } = req.params;
+      const { id } = request.params as { id: string };
       const params: RequestParams.Get = { id, index: INDEX.OPENDISTRO_ISM_CONFIG };
-      const { callWithRequest } = this.esDriver.getCluster(CLUSTER.DATA);
-      const results: SearchResponse<any> = await callWithRequest(req, "search", params);
-      return { ok: true, response: results };
+      const { callAsCurrentUser: callWithRequest } = this.esDriver.asScoped(request);
+      const results: SearchResponse<any> = await callWithRequest("search", params);
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: results,
+        },
+      });
```

Comment:
Rather than `response.custom`, you can use the given `response.ok` which will return a `statusCode` of `200` by default.

Reply:
Was using `response.ok` but resulted in a lot of type error. So changed back to `response.custom`. Thanks for the reminder.